### PR TITLE
Features/pagination

### PR DESCRIPTION
**Pull request by tomasvarga**

### CODE REVIEW

File: src/backend.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+const backend = async (res) => {
+  try {
+    const result = await fetch(res.url);
+    const data = await result.json();
+    const header = await result.headers;
```
Comment (pmadar) [inline src/backend.js]: `result.headers` is not an `async` function I think, right?

File: src/components/CardList.js [hunk 2]
```diff
@@ -3,44 +3,58 @@ import PropTypes from 'prop-types';
...
-        <PhotoCard
-          key={value.id}
-          title={value.title}
-          img={value.img}
-        />
-      )))}
-    </List>
+    <div>
+      <List style={{ flexDirection: isListView ? 'column' : 'row' }}>
+        {newData.map(value => (isListView ? (
```
Comment (pmadar) [inline src/components/CardList.js]: You don't need to compare it on every loop iteration.
What about this?
```js
{isListView
	? newData.map(value =>
		<LongCard ... />
	)
	: newData.map(value =>
		<PhotoCard ... />
	)
}
```

File: src/hoc/withInfinityScroll.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
```
Comment (pmadar) [inline src/hoc/withInfinityScroll.js]: duplicate of `src/components/hoc/withInfinityScroll.js`

File: src/App.js [hunk 4]
```diff
@@ -111,18 +102,21 @@ class App extends Component {
   async loadNewPage() {
     this.setState({ isError: false, isLoading: true });
     const { requiredItemsPerPage, actualPage, maxPages } = this.state;
-    const nextPage = this.updatePages(actualPage, maxPages);
-    if (nextPage > maxPages) {
-      this.setState({ isLoading: false });
-      return null;
+    const nextPage = this.getNextPage(actualPage, maxPages);
```
Comment (pmadar) [inline src/App.js]: You've forgot to rename `nextPage` to `actualPage`
```js
const actualPage = this.getNextPage(actualPage, maxPages);
```

Comment (tomasvarga) [reply src/App.js]: Thanks. As `actualPage `is already taken as a global variable, I'll change it to `currentPage`. It should be more understandable.

**Pull request merged**